{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Shahlab Pipeline Analysis Guidelines\n",
    "\n",
    "## Data Transfer\n",
    "\n",
    "### GSC aligned data\n",
    "1.. If you are on the project email list, you will receive an email from the GSC with a title similar to \" New FASTQ files available for SOW GSC-0297\", which will tell you the principal investigator who submitted these samples. The email contains an attachment with relevant information for sample identification and download.\n",
    "\n",
    "```\n",
    "POOLED DATA:\n",
    "library: IX4686\n",
    ".\n",
    ".\n",
    ".\n",
    "number_of_sublibraries: 3\n",
    "-------------------------\n",
    ">>>ALERT(S) DETECTED\n",
    "-------------------------\n",
    "LIBRARY\tINDEX\tFLOWCELL\tLANE\tANTIBODY\tUPPER_PROTOCOL\tLOWER_PROTOCOL\tTAXONOMY_ID\tGENOME_REF\tSEQ LENGTH\tPATIENT_ID\tEXTERNAL_ID\tORIGINAL_SOURCE\tSUBMITTED_SPIKE_IN_SEQUENCE\tSUBMITTED_SPIKE_IN_IDENTIFIER\tALERT_CODE\tALERT_NOTES\tBIO_QC_STATUS\tBIO_QC_COMMENTS\n",
    "A51910\tCCTTAG\tC9W6RANXX\t1\tNone\tStrand Specific Transcriptome 2.1\tIllumina Indexing\t9606\tHomo sapiens\t76\tSKBR3_KUTRsi_2\tSA_601\tNone\tNone\tNone\tNone\tPassed\t2016-08-30: (N/A -> Passed) Warning:0.076 portion of reads matching human ribosomal RNA is higher than threshold 0.05\n",
    "A60778\tCAAAAG\tC9W6RANXX\t1\tNone\tStrand Specific Transcriptome 2.2\tIllumina Indexing\t9606\tHomo sapiens\t76\tSKBR3_scramSix1\tSA902-R2\tSA_736\tNone\tNone\tNone\tNone\tPassed\t2016-08-30: (N/A -> Passed) Warning:0.074 portion of reads matching human ribosomal RNA is higher than threshold 0.05\n",
    "A60779\tCCAACA\tC9W6RANXX\t1\tNone\tStrand Specific Transcriptome 2.2\tIllumina Indexing\t9606\tHomo sapiens\t76\tSKBR3_KUTRsi_3\tSA903\tSA_737\tNone\tNone\tQC_21\tPortion of reads mapping to target species ribosomal RNA is higher than expected.\tFailed\t2016-08-30: (N/A -> Failed) Warning:0.125 portion of reads matching target mitochondria is higher than threshold 0.1;Failed:0.121 portion of reads matching human ribosomal RNA is higher than threshold 0.1\n",
    "```\n",
    " \t\t* library pooling id: IX4686\n",
    "\t\t* LIBRARY: A51910\n",
    "\t\t* EXTERNAL_ID: SKBR3_KUTRsi_2\n",
    "```\n",
    "WHOLE GENOME DATA:\n",
    "library: A62078\n",
    ".\n",
    ".\n",
    ".\n",
    "number_of_sublibraries: 1\n",
    "-------------------------\n",
    "LIBRARY\tINDEX\tFLOWCELL\tLANE\tANTIBODY\tUPPER_PROTOCOL\tLOWER_PROTOCOL\tTAXONOMY_ID\tGENOME_REF\tSEQ LENGTH\tPATIENT_ID\tEXTERNAL_ID\tORIGINAL_SOURCE\tSUBMITTED_SPIKE_IN_SEQUENCE\tSUBMITTED_SPIKE_IN_IDENTIFIER\tALERT_CODE\tALERT_NOTES\tBIO_QC_STATUS\tBIO_QC_COMMENTS\n",
    "A62078\tGCACTT\tH323YALXX\t1\tNone\tGenome Shotgun PCRFree 1.1\tIllumina Indexing\t9606\tHomo sapiens\t151\tHCT116\tSA988\tSA_829\tNone\tNone\tNone\tNone\tPassed\t2016-08-30: (N/A -> Passed) \n",
    "```\n",
    " \t\t* LIBRARY: A62078\n",
    " \t\t* EXTERNAL_ID: SA988\n",
    "2.. To transfer pooled exome and rnasomnieq data use the **library pooling id**, **LIBRARY** and **EXTERNAL_ID** information to make a *tab-separated* input file with the format, with each sample on one line:\n",
    "IX4686\tA51910\tSKBR3_KUTRsi_2\n",
    "\n",
    "3.. To transfer whole genome data, use the **LIBRARY** and **EXTERNAL_ID** information to make a *tab-separated* input file with the format, with each sample on one line:\n",
    "A62078\tSA988\n",
    "\n",
    "4.. Clone the stash repository <> to install the transfer script in your own folder on the GSC system.\n",
    "\n",
    "5.. Identify what the project name is for the samples in question, and make sure to batch samples by project in each input file. Samples will be soft linked in the relevant project folder in a samples/ subfolder (create the samples/ subfolder in the project folder on lustre if it doesn't already exist). The project name will be a command line input for the transfer script and must match the spelling of the /share/lustre/projects folder being written to.\n",
    "\n",
    "6.. ssh to xhost10 (so that the GSC folders can be accessed)\n",
    "\n",
    "7.. `cat my_inputfile.txt | sh make_transferBam_script_onBeast_merge_bwa-mem.sh -f -t \\<sample type>` to see the files that will be transferred.\n",
    "\n",
    "8.. `cat my_inputfile.txt | sh make_transferBam_script_onBeast_merge_bwa-mem.sh -t \\<sample type> -p \\<project>` to create the transfer script in your home directory on beast\n",
    "\n",
    "9.. ssh over to beast\n",
    "\n",
    "10.. Open a screen.\n",
    "\n",
    "11.. `sh transferScript_YYYY_MM_DD.sh` to start sample transfer\n",
    "\n",
    "12.. Sample files (bam, bai, flagstats, bamstats, fastq) should be transferred to the /share/lustre/archive/ folder and have the folder structure /share/lustre/archive/_sample\\_id_/illumina_tech/libID/_aligner_\\_aligned for bam files and /share/lustre/archive/_sample\\_id_/illumina\\__tech_/_libID_/sequence/ for fastq files\n",
    "\n",
    "### Collaborator data (non-GSC)\n",
    "1. Collect information about samples\n",
    "  * platform used to sequence data\n",
    "  * alignment method, duplicates marked, genome ascension number (hg19, mm10, etc)\n",
    "  * an existing master sample list for this collaborator/project\n",
    "2. Create meaningful sample ids for each sample, using prefixes that correspond to the sample submitter or project name, with a 4-digit number starting at 0001.. For a small number of samples for a small project, use external ids provided, with underscores separating relevant information.\n",
    "3. Because there is likely no library id for collaborator samples, just use the sample id as the library id when transferring samples to the archive.\n",
    "4. Sample files (bam, bai, flagstats, bamstats, fastq) should be rsync transferred to the /share/lustre/archive/ folder and have the folder structure /share/lustre/archive/_sample\\_id_/illumina_tech/libID/_aligner_\\_aligned for bam files and /share/lustre/archive/_sample\\_id_/illumina\\__tech_/_libID_/sequence/ for fastq files\n",
    "\n",
    "### Transfer to shahlab/archive/ or extscratch/shahlab/archive/ on the GSC\n",
    "1. When transferring data to shahlab/archive/, at the very least, rename the files with the sample id prepending the file name (for bam and bai files) and rsync to a folder named with the sample id, then rsync the folder strtucture from the GSC, as desired (eg. 75nt/hg19a_jg-e69/bwa-0.5.7/)\n",
    "2. To transfer data to extscratch/shahlab/archive/, there is an archive manager available to transfer files from lustre. There is also much more limited space on extscratch/ than on shahlab/\n",
    " \t* calculate how much space the files take up using something like: \n",
    " \t```ls -hs LIST.OF.FILES | awk -F'G' 'BEGIN {sum=0} /^[ 0-9.]*G/ {sum+=$1} END {print sum\"G\"}'```\n",
    " \t* login to the gsc & check if there is enough space to transfer these bam files `du -h /genesis/extscratch/shahlab/`\n",
    " \t* Create input file to /genesis/extscratch/shahlab/archive/manager/code/projects/*tsv\n",
    "    *Input file should contain the following:\n",
    "    *header labelled: \"bccrc_file_path\"\n",
    "    *contain a list of files to be transferred as they appear in /share/lustre/archive on beast\n",
    " \t* ssh thost05\n",
    " \t* open screen\n",
    " \t* source /genesis/extscratch/shahlab/archive/manager/bash_profile\n",
    "\t* `python /genesis/extscratch/shahlab/archive/manager/code/manager.py /genesis/extscratch/shahlab/archive/manager/code/config.yaml --get_bam_index`\n",
    "\n",
    "When files are transferred, they will be in the same archive folder structure on extscratch as they are in the lustre archive folder.\n",
    "\n",
    "\n",
    "## Ticket Setup for New Analyses on [Jira](https://www.bcgsc.ca/jira/login.jsp)\n",
    "1.. Create a master Jira ticket under the correct project name with all samples in the project listed in a table in the ticket Description.\n",
    "For example:\n",
    "\n",
    "| SA_ID | LIbID |Downloaded | GSNAP | MISO | CUFFLINKS | CUFFLINKS_MATRIX |\n",
    "|-\t| -| -| -| -| -| -|\n",
    "|SA494X4\t|A60788| | | | | | \n",
    "|SA495X5\t|A60790| | | | | | \n",
    "|SA496X2\t|A60782| | | | | | \n",
    "\n",
    "2. Create a sub-ticket for each task (hint: the analysis list for this sample set is given as a header in the table) and add all relevant collaborators as watchers.\n",
    " \t* From the **More** dropdown menu > select **Create Sub-task**\n",
    " \t* Copy the SampleID, LibID and relevant analysis type to the new sub-ticket and create a similar table to track analyses on that ticket. Add watchers again to this ticket.\n",
    " \t* When the task is started, select the **Start Progress** button to commence \n",
    " \t* When the sub-task is completed, select **Resolve Issue** then have the Reporter or another supervisor approve the closure of the ticket. \n",
    "\n",
    "## Pipeline Runs\n",
    "1. Select a folder within your own directory and clone the relevant pipeline from shahlab [Stash](https://svn.bcgsc.ca/bitbucket/projects) into it.\n",
    "2. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
